packages/vite/src/node/ssr/fetchModule.ts

```diff
@@ -41,21 +42,22 @@ export async function fetchModule(
       resolve: { dedupe, preserveSymlinks },
       root,
       ssr,
-    } = server.config
-    const overrideConditions = ssr.resolve?.externalConditions || []
+    } = environment.config
+    const externalConditions = ssr.resolve?.externalConditions || []
```

Comment:
In relate to https://github.com/vitejs/vite/pull/16129/#discussion_r1546009721, should this take `environment.options.resolve` instead of `environment.config.resolve` or  `environment.config.ssr.resolve`?

Reply:
Yes, `config.ssr.resolve` shouldn't be used at all in the code. There are some cases where I can't yet replace them because the environment is not properly in the context.